### PR TITLE
Magazine Audit - 12g, .30-06, .338LM, .38 Super, .410, .460 Rowland, .50BMG

### PR DESCRIPTION
**Pull request by ANickelN**
#### Summary
Balance "Audit 12g, .30-06, .338LM, .38 Super, .410, .460 Rowland, .50BMG Magazines"

#### Purpose of change

Magazines are the wrong size and weight, this PR has better sizes and weights.

#### Describe the solution

Give these magazines the correct dimensions and weight.

#### Describe alternatives you've considered

There is no alternative.

#### Testing

TBD

#### Additional context

This PR is part of issue https://github.com/CleverRaven/Cataclysm-DDA/issues/70784

Auditing information available below as per my post in that issue:

### 12 Gauge

8rd Magazine
9 1/2" long
3" wide
1 3/8" thick
Brass diameter of 12g is 0.8090 in 

Saiga 12, 10rd Polymer
KS12MAG10 

From Manufacturer:
Weight	0.75 lbs
Dimensions	3.0 × 1.375 × 11.75 in

Width didn't match in ImageJ, so bumped down to 3cm
29.845 cm * 7.62 cm * 3.0 cm = 682 ml
0.75lb = 340 g

Saiga 12, 30rd Drum -> 20rd drum (There are no 30rd Saiga 12 drums, there is one for the .410 version)
My guess is that when it was added someone briefly glanced at Wikipedia, saw "30rd Drum" there and didn't realize it was only for the .410 version.
Modeled on the Promag SAI-A6 20rd Drum Magazine (Can only fit 2.75" shells)

21.75cm tall (feeding portion, not going to use for calculation)
18cm diameter * 8cm front-to-back 
2036mL
Used 0.438g/mL for comparison weight, bumped up by 1.2x because more springs involved with drum.
1070g

20rd shotshell belt
This one is mostly guesswork, this is a weird item.
12g shell 5.8738 cm long * diameter of 2.05486 cm = 19.48ml
19.48ml * 20shells * 1.1(cloth)
429ml when filled
35 ml
41g (weighed a few cotton socks)

6rd Shotgun Speedloader
Shotgun speedloaders based on https://www.syringeshotgunspeedloader.com/

Took the volume of a 2.75" shell, and multiplied by 6, then *1.1 to account for container
19.48 * 6 * 1.1 = 129 ml
12.9 g

8rd Shotgun Speedloader 
19.48 * 8 * 1.1 = 171 ml
17.1 g

Previous PR for these tubes had shotgun shells set at 70ml for reference, which is way too large.
Weight was gotten by multiplying volume by 0.1, these are mostly hollow tubes.

M26 Mass Magazines
TC 3-22.12 used as reference for ImageJ

3rd
11.75cm tall * 7.75cm front to back * 2.8cm wide
255ml
Used 0.438g/mL for comparison weight: 112g

5rd
15.75cm tall * 7.75cm front to back * 2.8cm wide
342ml
Used 0.438g/mL for comparison weight: 150g

### .30-06.

.30-06 Clip
Copied from 7.62x54R Clip Values

BAR 20rd Magazine (From schematics, thanks to Tonkatsu)
5.7" tall * 3.437" front-to-back * .995"
319ml
184g

BAR 30rd Magazine (ImageJ)
7.41" tall * 3.437" * .995"
415ml
239g

### .338 Lapua Magnum

Accurate Mag .338 Lapua 5 round AICS magazine
https://riflemags.co.uk/accurate-mag-338-lapua-5-round-aics-magazine-single-stack/

10.4cm tall * 9.5cm front-to-back * 1.7cm wide: using ImageJ with cartridge as comparison
Afterwards I found actual dimensions.  (Helps if you scroll down the page instead of assuming they'll be missing)
9.2964 cm tall * 9.4361 cm front-to-back * 1.86944 cm wide : From vendor
168ml was my ImageJ value, 164ml was actual value!  Only 2.4% error.
215g per the vendor.

10rd Accuracy International 6434 Magazine

9.5cm tall * 9.4361 cm * 2.6cm wide: 
233mL per ImageJ
Used ratio with 5rd: 1.2797g/ml
298g

10rd Barrett MRAD Magazine (Polymer Magazine)
195 Grams per UK vendor
ImageJ'd:
10 cm tall * 9.4361 cm * 2.6 cm
245ml

### .38 Super

AF2011A1 16rd Magazine (Really weird, so working off of the 9rd .38 Super magazine.
10 cm tall * 2.5 cm front-to-back * 2.4 cm
60mL
113g (Ratio of g/ml from the 9rd magazine)

1911 9rd Magazine audited with ImageJ
11 cm tall * 2.5 cm front-to-back * 1.2 cm 

33mL
62g from a similar 1911 .45 7rd Magazine

### .410

Another rare gun with the Saiga .410, most of the ones that were sold were 12g from what I understand.

Saiga .410 10rd
Based on the SGM Tactical Saiga 410 Gauge 10-Round Polymer Magazine
ImageJ used with 3" shell as reference
19.7cm tall * 8.6cm front to back * 2.4cm wide = 407mL
Used comparison method for weight, with a similar size polymer AK mag. 170g/388mL = 0.438g/mL
Weight of 178g

Saiga .410 30rd Drum
This one is pretty crude because I couldn't get a perfectly head on picture.
Based on Promag 30rd Drum
ImageJ used with 3" shell as reference
11.5cm tall * 9cm front to back * 7.5cm wide
777mL
Used comparison method for a weight of 340g

### .460 Rowland/.45 1911s

These magazines are for .45 and .460 1911

7rd looks correct

10rd
Dimensions of magazine from a friend.
15.5cm * 3.4798cm * 1.3716
74 ml
79 g

### .50BMG

10rd M107A1 Magazine

15 cm tall * 14.3 cm front-to-back * 4 cm
858ml
Left weight from previous Previous

10rd AS-50 Magazine -> 5rd AS-50 Magazine (The AS-50 doesn't have a 10rd magazine)
Had to base this one on the AX50 mag, which is similar.

13.25cm tall * 14.3 cm front-to-back * 2.4 cm wide
455 ml
Used ratio formula for weight from the Barett mag. 0.847g/ml
386 g

5rd Tac-50 Magazine
I just copied the stats from the AS-50 magazine.

### CODE REVIEW

File: data/json/items/magazine/338lapua.json [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+    "//": "Audited with dimensions/weight from vendor.",
     "looks_like": "stanag30",
     "type": "MAGAZINE",
     "name": { "str": "AI .338 Lapua 5-round magazine" },
     "description": "A single-stack, 5-round steel box magazine, produced by the British precision rifle company, Accuracy International.  Magazines such as this one have been regarded as the gold standard for marksman rifle magazines, with this particular one being compatible with both the .338 Lapua and .300 Winchester Magnum cartridges.",
-    "weight": "78 g",
-    "volume": "250 ml",
+    "weight": "215 g",
+    "volume": "164 ml",
+    "longest_side": "930 mm",
```
Comment (FlyingSpaceRhino) [inline data/json/items/magazine/338lapua.json:11]: should be 93mm, 1cm=10mm not 100mm

File: data/json/items/magazine/338lapua.json [hunk 2]
```diff
@@ -18,12 +20,14 @@
...
     "looks_like": "stanag30",
     "type": "MAGAZINE",
     "name": { "str": "AI .338 Lapua 10-round magazine" },
-    "description": "A single-stack, 10-round steel box magazine, produced by the British precision rifle company, Accuracy International.  Magazines such as this one have been regarded as the gold standard for marksman rifle magazines, with this particular one being compatible with both the .338 Lapua and .300 Winchester Magnum cartridges.",
-    "weight": "160 g",
-    "volume": "400 ml",
+    "description": "A double-stack, 10-round steel box magazine, produced by the British precision rifle company, Accuracy International.  Magazines such as this one have been regarded as the gold standard for marksman rifle magazines, with this particular one being compatible with both the .338 Lapua and .300 Winchester Magnum cartridges.",
+    "weight": "298 g",
+    "volume": "233 ml",
+    "longest_side": "950 mm",
```
Comment (FlyingSpaceRhino) [inline data/json/items/magazine/338lapua.json:30]: should be 95mm

**Review by FlyingSpaceRhino — CHANGES_REQUESTED** [review]

**Comment by Hyperseeker** [comment]
Absolutely loving the research notes in-between listing dimensions. :D

**Review by FlyingSpaceRhino — CHANGES_REQUESTED** [review]

File: data/json/items/magazine/338lapua.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ANickelN) [reply data/json/items/magazine/338lapua.json]: Thanks for catching this, forgot the decimal!

File: data/json/items/magazine/338lapua.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ANickelN) [reply data/json/items/magazine/338lapua.json]: Thanks again, forgot the decimal.

**Comment by ANickelN** [comment]
> Absolutely loving the research notes in-between listing dimensions. :D

Yeah some of these are a massive pain to find so it only makes sense to try and leave my work so other people can check it.

**Hardest**
Magazines for rare firearms 
(Information on the firearm itself is hard to find, magazines is ridiculous.  Best example in this is the AS-50, doesn't seem to have ever actually been produced beyond a prototype)
Magazines with weird shapes like drums 
(Usually little/no information on dimensions, and pictures are usually poor quality.  Not as bad on WW2 firearms issued with drums, but still hard)

**Easier**
Magazines for firearms used in military service (Usually blueprints if older, otherwise there are good pictures)
Magazines for common firearms (Rarely find blueprints, but images are usually good for use with ImageJ)

I know standards for addition of firearms was laxer in the past, and future additions have certain criteria so I'm glad I will be seeing less of "Yes, this is the Bipplenopper 2.45, they only produced 1 of them ever and it was a total failure, I'm adding it to the game."  There's plenty of really common firearms in the US that are still not in the game.

**Pull request merged**